google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java

```diff
@@ -332,18 +339,29 @@ private void maybeWaitForInflightQuota() {
     long start_time = System.currentTimeMillis();
     while (this.inflightRequests >= this.maxInflightRequests
         || this.inflightBytes >= this.maxInflightBytes) {
-      try {
-        inflightReduced.await(100, TimeUnit.MILLISECONDS);
-      } catch (InterruptedException e) {
-        log.warning(
-            "Interrupted while waiting for inflight quota. Stream: "
-                + streamName
-                + " Error: "
-                + e.toString());
+      if (this.limitExceededBehavior == FlowController.LimitExceededBehavior.ThrowException) {
+        throw new StatusRuntimeException(
+            Status.fromCode(Code.RESOURCE_EXHAUSTED)
+                .withDescription(
+                    "Exceeds client side inflight buffer, consider add more buffer or open more connections."));
+      } else if (this.limitExceededBehavior == FlowController.LimitExceededBehavior.Ignore) {
```

Comment:
Wouldn't be the ignore option just a break; at this point?

Reply:
L735, by default it is Block.

Reply:
https://github.com/googleapis/java-bigquerystorage/pull/1643

Reply:
What I meant was, that the actual meaning of LimitExceededBehavior.Ignore is, that this inflight quota counting is simply ignored. So the StreamWriter just continues sending. But this option is probably not that relevant for productive setups.
```suggestion
      } else if (this.limitExceededBehavior == FlowController.LimitExceededBehavior.Ignore) {
        break;
      }
```

Reply:
I tend to think we shouldn't allow it...